packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -622,70 +757,31 @@ export function getPathForPackageManager({
    */
   yarnNodeLinker: string | undefined;
 } {
-  let detectedLockfile: string | undefined;
-  let detectedPackageManager: string | undefined;
-  let pathValue: string | undefined;
-  let yarnNodeLinker: string | undefined;
-  const oldPath = env.PATH + '';
-  const npm7 = '/node16/bin-npm7';
-  const pnpm7 = '/pnpm7/node_modules/.bin';
-  const pnpm8 = '/pnpm8/node_modules/.bin';
-  const bun1 = '/bun1';
-  const corepackEnabled = env.ENABLE_EXPERIMENTAL_COREPACK === '1';
-  if (cliType === 'npm') {
-    if (
-      typeof lockfileVersion === 'number' &&
-      lockfileVersion >= 2 &&
-      (nodeVersion?.major || 0) < 16 &&
-      !oldPath.includes(npm7) &&
-      !corepackEnabled
-    ) {
-      // npm 7
-      pathValue = npm7;
-      detectedLockfile = 'package-lock.json';
-      detectedPackageManager = 'npm 7+';
-    }
-  } else if (cliType === 'pnpm') {
-    if (
-      typeof lockfileVersion === 'number' &&
-      lockfileVersion === 5.4 &&
-      !oldPath.includes(pnpm7) &&
-      !corepackEnabled
-    ) {
-      // pnpm 7
-      pathValue = pnpm7;
-      detectedLockfile = 'pnpm-lock.yaml';
-      detectedPackageManager = 'pnpm 7';
-    } else if (
-      typeof lockfileVersion === 'number' &&
-      lockfileVersion >= 6 &&
-      !oldPath.includes(pnpm8) &&
-      !corepackEnabled
-    ) {
-      // pnpm 8
-      pathValue = pnpm8;
-      detectedLockfile = 'pnpm-lock.yaml';
-      detectedPackageManager = 'pnpm 8';
-    }
-  } else if (cliType === 'bun') {
-    if (!oldPath.includes(bun1) && !corepackEnabled) {
-      // Bun 1
-      pathValue = bun1;
-      detectedLockfile = 'bun.lockb';
-      detectedPackageManager = 'Bun';
-    }
-  } else {
-    // Yarn v2 PnP mode may be activated, so force "node-modules" linker style
-    if (!env.YARN_NODE_LINKER) {
-      yarnNodeLinker = 'node-modules';
-    }
-  }
-  return {
-    detectedLockfile,
-    detectedPackageManager,
-    path: pathValue,
-    yarnNodeLinker,
+  const overrides = getPathOverrideForPackageManager({
+    cliType,
+    lockfileVersion,
+    nodeVersion,
+    env,
+  });
+
+  const alreadyInPath = (newPath: string) => {
+    const oldPath = env.PATH ?? '';
+    return oldPath.split(path.delimiter).includes(newPath);
   };
+
+  switch (true) {
+    case cliType === 'yarn' && !env.YARN_NODE_LINKER:
+      return { ...overrides, yarnNodeLinker: 'node-modules' };
+    case alreadyInPath(overrides.path ?? ''):
+      return {
+        detectedLockfile: undefined,
+        detectedPackageManager: undefined,
+        path: undefined,
+        yarnNodeLinker: undefined,
+      };
+    default:
+      return { ...overrides, yarnNodeLinker: undefined };
+  }
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏ifs over switches?

The switch-true pattern is harder to understand than the equivalent if-return or if-else, in my opinion. I suggest changing this, but it's up to you.